### PR TITLE
Add end-to-end encryption when creating session.

### PR DESCRIPTION
**Pull request by SecondeJK**
This PR adds the end to end encryption flag when starting an OpenTok Session.

## Description
See the docs for more context:
https://tokbox.com/developer/guides/end-to-end-encryption/

## Motivation and Context
New features being rolled into our SDKs

## How Has This Been Tested?
Added new test and modified existing one to make sure defaults are read properly.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/OpenTok/OpenTok.php [hunk 1]
```diff
@@ -166,6 +165,9 @@ public function generateToken($sessionId, $options = array())
     *    (either automatically or not), you must set the <code>mediaMode</code> key to
     *    <code>MediaMode::ROUTED</code>.</li>
     *
+    *    <li><code>'e2ee'</code> (Boolean) &mdash; Whether to enable end-to-end encryption for the
+    *    OpenTok session.</code>.</li>
```
Comment (jeffswartz) [inline src/OpenTok/OpenTok.php:169]:
```suggestion
    *    <li><code>'e2ee'</code> (Boolean) &mdash; Whether to enable
    *    <a href="https://tokbox.com/developer/guides/end-to-end-encryption">end-to-end encryption</a>
    *    for the OpenTok session.</li>
```

File: src/OpenTok/Session.php [hunk 2]
```diff
@@ -148,4 +158,12 @@ public function generateToken($options = array())
     {
         return $this->opentok->generateToken($this->sessionId, $options);
     }
+
+    /**
+     * @return mixed
```
Comment (jeffswartz) [inline src/OpenTok/Session.php:163]: Should this be `@return bool`? And can we please add a description:

```suggestion
    /**
     * Whether <a href="https://tokbox.com/developer/guides/end-to-end-encryption">end-to-end encryption</a>
     * is set for the session.
     *
     * @return mixed
```

**Review by jeffswartz — CHANGES_REQUESTED** [review]
Should we rev the version (in Client.php and elsewhere) to 4.12.0?

**Comment by SecondeJK** [comment]
@jeffswartz I've not bumped the hardcoded version number as #328 will rewrite how this is handled (actually addresses a TODO from 2014 on dynamic user agent string handling)

File: src/OpenTok/OpenTok.php [hunk 3]
```diff
@@ -166,6 +165,10 @@ public function generateToken($sessionId, $options = array())
     *    (either automatically or not), you must set the <code>mediaMode</code> key to
     *    <code>MediaMode::ROUTED</code>.</li>
     *
+    *    <li><code>'e2ee'</code> (Boolean) &mdash; Whether to enable
+    *    <a href="https://tokbox.com/developer/guides/end-to-end-encryption">end-to-end encryption</a>
+    *    for the OpenTok session.</li>
```
Comment (jeffswartz) [inline src/OpenTok/OpenTok.php:170]:
```suggestion
    *    for a routed session.</li>
```

**Review by jeffswartz — COMMENTED** [review]
Please add a commit to rev the minor version.

**Review by jeffswartz — APPROVED** [review]

**Pull request merged**